analysis/analysis.py

```diff
@@ -446,6 +450,203 @@ def test_word_time_relations(self):
+
+
+def compute_view_time_per_response(student_data):
+    """
+    Compute the total reread times for each response
+    :param student_data: list, student response dicts
+    :return:
+    wanted_dict_ad - Dictionary ('ad context response in lowercase': [total reading times for
+        each user])
+    wanted_dict_ss - Dictionary ('short story context response in lowercase': [total reading
+        times for each user])
+    """
+
+    wanted_dict_ad = defaultdict(list)
+    wanted_dict_ss = defaultdict(list)
+
+    for entry in student_data:
+        if entry['question'] == 'In one word, how does this text make you feel?':
+            response = entry['response'].lower()
+            views = sum(entry['views'])
+
+            if entry['context'] == 'This is an ad.':
+                wanted_dict_ad[response].append(views)
+
+            elif entry['context'] == 'This is actually a short story.':
+                wanted_dict_ss[response].append(views)
+
+    return {'This is an ad.': wanted_dict_ad, 'This is actually a short story.': wanted_dict_ss}
+
+
+def get_common_elements(ad_response_times, ss_response_times):
+    """
+    make a dictionary of the common elements between two dictionaries and averages them
+    :param ad_response_times: the first dictionary to compare
+    :param ss_response_times: the second dictionary to compare
+    :return: one dictionary
+    """
+    # Start making a combined dictionary that does not separate based on context
+    wanted_dict_combined = {}
+    for word in ad_response_times.keys():
+        if word not in wanted_dict_combined.keys():
+            wanted_dict_combined[word] = ad_response_times[word]
+
+    # Continue building the combined dictionary with short story dictionary
+    for word in ss_response_times.keys():
+        # Average current response with new data
+        if word in wanted_dict_combined.keys():
+            wanted_dict_combined[word] = (wanted_dict_combined[word] + ss_response_times[word]) / 2
+
+        # Add the new data
+        else:
+            wanted_dict_combined[word] = ss_response_times[word]
+
+    return wanted_dict_combined
+
+
+def reread_time_difference(ad_response_times, ss_response_times):
+    """
+    make a dictionary of which context had a longer reread time for each response and the difference
+    :param ad_response_times: dictionary of ad response times
+    :param ss_response_times: dictionary of ss response times
+    :return: dict{response:[context,time]}
+    """
+    # Make a dictionary that gives the name of the context with the longer average response time
+    # and difference between two contexts if a response is common to both contexts
+    wanted_dict_difference = {}
+
+    for word in ad_response_times.keys():
+        if word in ss_response_times.keys():
+            if ad_response_times[word] > ss_response_times[word]:
+                wanted_dict_difference[word] = ["ad", ad_response_times[word]
```

Comment:
The linebreak before the ```-``` makes this hard to read. Please either store it in a new var or fix the formatting.